### ISSUE TITLE
Markdown code block injection can leak to the surrounding context

### ISSUE THREAD

**Issue by kirawi** [helix-editor/helix#9678]
### Summary

If the code block's closing backticks have trailing space, the injected syntax highlighting will apply to the text outside of the code block.

### Reproduction Steps

Open a markdown buffer with the contents:

```
'''c
''' 
int i = 0;
```
where the quotes are replaced with backticks.

I expected this to happen:

`int i = 0;` should be highlighted as markdown

Instead, this happened:

It is highlighted as C

### Platform

Linux/macOS

### Terminal Emulator

Kitty 0.30.1

### Installation Method

Source

### Helix Version

990378a

**Comment by the-mikedavis** [comment]
It seems like this is because the markdown grammar only treats <code>```</code> followed by newlines as a terminator for a codefence. I'm not sure if that's intentional though

**Comment by pascalkuthe** [comment]
yeah I am pretty sure that is the reason. That may genuinely be the way markdown parsers behave. You can report that upstream to figure out if its intentional or not but we can't change this downstream

**Comment by kirawi** [comment]
According [to the spec](https://github.github.com/gfm/#fenced-code-blocks) it references:

> The closing code fence may be indented up to three spaces, and may be followed only by spaces, which are ignored.

**Comment by haitrungle** [comment]
I think this is fixed in the latest version?

**Comment by the-mikedavis** [comment]
Looking at https://github.com/tree-sitter-grammars/tree-sitter-markdown/issues/135 it looks like this should be fixed upstream now. We'll need to update https://github.com/helix-editor/helix/blob/2ee11a0a9d9b9951b5b64b54be05379cc030230b/languages.toml#L1787-L1789

and markdown_inline

**Comment by the-mikedavis** [comment]
Fixed in #14814
